tools/migration/rdbms/migrationlib/version.go

```diff
@@ -6,11 +6,68 @@
 package migrationlib
 
 import (
+	"fmt"
+
 	"database/sql"
 	"github.com/pressly/goose"
 )
 
-// DBVersion returns the current version of the database schema
-func DBVersion(db *sql.DB) (int64, error) {
-	return goose.GetDBVersion(db)
+// queryExecutor represents the MCD interface between sql.Db and sql.Tx
+// objects that is sufficient to retrieve the db version number.
+type queryExecutor interface {
+	Query(query string, args ...interface{}) (*sql.Rows, error)
+}
+
+// DBVersion returns the current version of the database schema.
+// The library exposes a GetDBVersion function which however requires
+// a *sql.Db object. Several contest flows work instead within a
+// transaction al context, so we need to be able to extract the version
+// in a way that is agnostic to the library.
+func DBVersion(db queryExecutor) (uint64, error) {
```

Comment:
I could not find a way to avoid this, unfortunately. I am checking with upstream library maintainer if they accept PRs to make the library more flexible in terms of what it supports (`sql.Tx` vs `sql.Db` storage). Until then, we need stay on this stopgap. If nothing moves upstream I think we might want to consider building our own small library.